code/game/machinery/adv_med.dm

```diff
@@ -185,6 +261,47 @@
 			A.loc = src.loc
 		qdel(src)
 
+
+/obj/machinery/body_scanconsole
+	var/obj/machinery/bodyscanner/connected
+	var/known_implants = list(/obj/item/weapon/implant/chem, /obj/item/weapon/implant/death_alarm, /obj/item/weapon/implant/loyalty, /obj/item/weapon/implant/tracking)
+	var/delete
+	var/temphtml
+	name = "body scanner console"
+	icon = 'icons/obj/Cryogenic2.dmi'
+	icon_state = "body_scannerconsole"
+	density = 1
+	anchored = 1
+	var/orient = "LEFT"
+
+/obj/machinery/body_scanconsole/New()
+	..()
+	spawn(5)
+		if(orient == "RIGHT")
+			icon_state = "body_scannerconsole-r"
+			src.connected = locate(/obj/machinery/bodyscanner, get_step(src, EAST))
+		else
+			src.connected = locate(/obj/machinery/bodyscanner, get_step(src, WEST))
+	return
+
+/obj/machinery/body_scanconsole/update_icon()
```

Comment:
Modular icon states like these can be constructed instead of done reeeally manually like this
icon_state = "body_scannerconsole[stat & NOPOWER? null : "-p"][orient == "LEFT" ? null : "-r"]"